### PR TITLE
 [Fabric] Make surface presenter manage start of surface

### PR DESCRIPTION
**Pull request by zhongwuzw**
## Summary

1. Start surface when first run, before, when we first load js bundle, `_startAllSurfaces` would not be called because `bridge == _batchedBridge`.
2. I think this fixes https://github.com/facebook/react-native/issues/23910, so I add the `return` in swizzle method to check. 

cc @shergin.

## Changelog

[iOS] [Fixed] - Make surface presenter manage start of surface

## Test Plan

N/A.

### CODE REVIEW

File: React/Fabric/Surface/RCTFabricSurface.mm [hunk 1]
```diff
@@ -67,8 +65,7 @@ - (BOOL)start
   if (![self _setStage:RCTSurfaceStageStarted]) {
     return NO;
   }
-
-  [_surfacePresenter startSurface:self];
+  [_surfacePresenter registerSurface:self];
```
Comment (shergin) [inline React/Fabric/Surface/RCTFabricSurface.mm]: I definitely like this part. This is how it should be IMHO.

File: React/Fabric/Surface/RCTFabricSurface.mm [hunk 2]
```diff
@@ -55,8 +55,6 @@ - (instancetype)initWithSurfacePresenter:(RCTSurfacePresenter *)surfacePresenter
     _touchHandler = [RCTSurfaceTouchHandler new];
 
     _stage = RCTSurfaceStageSurfaceDidInitialize;
```
Comment (shergin) [inline React/Fabric/Surface/RCTFabricSurface.mm]: 👍

File: RNTester/RNTester/AppDelegate.m [hunk 3]
```diff
@@ -77,6 +77,7 @@ - (BOOL)application:(__unused UIApplication *)application didFinishLaunchingWith
 #ifdef RN_FABRIC_ENABLED
 // FIXME: remove when resolved https://github.com/facebook/react-native/issues/23910
 - (void)handleJavaScriptDidLoadNotification:(__unused NSNotification*)notification {
+  return;
```
Comment (shergin) [inline RNTester/RNTester/AppDelegate.m]: And please change this.

**Comment by zhongwuzw** [comment]
@shergin How about this, we add a stage check before start surface, not check bridge's equality.

File: RNTester/RNTester/AppDelegate.mm [hunk 4]
```diff
@@ -98,15 +86,6 @@ - (BOOL)application:(__unused UIApplication *)application didFinishLaunchingWith
   return YES;
 }
 
-#ifdef RN_FABRIC_ENABLED
```
Comment (zhongwuzw) [inline RNTester/RNTester/AppDelegate.mm]: I removed all these hack code.

**Pull request closed without merging**